src/exoticatechnologies/modifications/exotics/impl/GuardianShield.kt

```diff
@@ -0,0 +1,711 @@
+        Utilities.takeItemQuantity(fleet.cargo, ITEM, 1f)
+        return true
+    }
+
+    override fun modifyToolTip(
+            tooltip: TooltipMakerAPI,
+            title: UIComponentAPI,
+            member: FleetMemberAPI,
+            mods: ShipModifications,
+            exoticData: ExoticData,
+            expand: Boolean
+    ) {
+        StringUtils.getTranslation(key, "longDescription")
+                .format("flux_method", lazyFluxMethodString)
+                .format("flux_effects", lazyFluxEffectString)
+                .addToTooltip(tooltip, title)
+    }
+
+    override fun applyToShip(id: String, member: FleetMemberAPI, ship: ShipAPI, mods: ShipModifications, exoticData: ExoticData) {
+        super.applyToShip(id, member, ship, mods, exoticData)
+
+        originalShip = ship
+        originalOwnersShield = ship.shield
+        // First thing's first - disable the ship's shield
+        ship.setShield(ShieldAPI.ShieldType.NONE, 0f, 0f, 0f)
+
+        // Secondly, give the ship a new subsystem to summon a drone hosting the Guardian Shield that will follow the ship around
+        val subsystem = GuardianShieldDrone(ship)
+        subsystem.setDronesToSpawn(1)
+        MagicSubsystemsManager.addSubsystemToShip(ship, subsystem)
+    }
+
+    override fun onDestroy(member: FleetMemberAPI) {
+        super.onDestroy(member)
+
+        originalShip.setShield(
+                originalOwnersShield.type,
+                originalOwnersShield.upkeep,
+                originalOwnersShield.fluxPerPointOfDamage,
+                originalOwnersShield.arc
+        )
+    }
+
+    inner class GuardianShieldDrone(ship: ShipAPI) : MagicDroneSubsystem(ship) {
+
+        var engine: CombatEngineAPI = Global.getCombatEngine()
+        private var drone: ShipAPI? = null
+        private val tracker = IntervalUtil(0.3f, 0.5f)
+        private val shieldController = ShieldController()
+
+        override fun onActivate() {
+            super.onActivate()
+            if (drone == null) {
+                drone = spawnDrone()
+                drone?.let { it.shield.toggleOn() }
+            } else {
+                drone?.let { it.shield.toggleOn() }
+            }
+        }
+
+
+        override fun getBaseActiveDuration(): Float = 0f
+
+        override fun getBaseCooldownDuration(): Float = 0f
+
+        override fun getDisplayText(): String = "Guardian Shield"
+
+        override fun getDroneVariant(): String = "exotica_guardianshield_standard"
+
+        override fun targetOnlyEnemies(): Boolean = false
+
+        override fun requiresTarget(): Boolean = false
+
+        override fun shouldActivateAI(amount: Float): Boolean {
+            if (engine.isPaused) {
+                return false
+            }
+            if (drone == null) {
+                drone = spawnDrone()
+            }
+            drone?.let {
+                if (!it.isAlive && ship.isAlive) {
+                    return false
+                }
+            }
```

Comment:
the drone is untargettable and undestructable, so this isn't a concern anymore.

Reply:
can leave it in, it could be useful for derivatives of this class, not very useful for this class except before the drone is spawned...

just let it stay there.

Reply:
resolving.